src/bin/grid.rs

```diff
@@ -43,22 +43,22 @@ mod example {
         window.set_application(application);
         let grid: Grid = builder.get_object("grid").expect("Couldn't get grid");
         let button6: Button = builder.get_object("button6").expect("Couldn't get button6");
-        button6.connect_clicked(clone!(grid, button6 => move |_| {
-            let height = grid.get_cell_height(&button6);
+        button6.connect_clicked(clone!(grid => move |button| {
```

Comment:
Is the button inside the grid? If so then this still leads to a reference cycle (use a weak ref for the grid)

Reply:
What do you mean by weak ref? The `WeakRef` type or something else?

Reply:
Yes, call `downgrade()` before, pass the `WeakRef` into the closure, call `upgrade()` inside the closure and if it returns `None` simply do nothing